### ISSUE TITLE
TODO list

### ISSUE THREAD

**Issue by YutackPark** [MDIL-SNU/SevenNet#86]
TODO list

-----------  on 'refactor' branch && v0.10.0 --------------
- [x] Hide `type_map`
- [x] Hide toggle grad things
- [x] Refactoring or from scratch: `scripts/processing_dataset.py` and `train/dataset.py`
- [x] Implement `universal` keyword for `chemical_species` key
- [x] Remove <cuda_runtime.h> from `pair_e3gnn.cpp`
- [x] Refactoring sevenn_inference (I bet writing from scratch is better)
- [x] #99, add model argument for calculator
- [x] #92
- [x] Remove deployed models in `pretrained_potentials`
- [x] Merge above changes(v0.10.0) (don't forget #73)

------------ Urgent -----------------
- [x] Notebook tutorial
- [x] Modern test code + github actions
- [ ] documentation
- [x] #71

----------- Future -------------
- [x] on-the-fly graph build training routine  # experimentally implemented in ase_db branch
- [ ] Accelerated training with graph [padding](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.transforms.Pad.html) and `torch.compile`
- [x] Modal select inputs for LAMMPS
- [ ] `tag`-wise weights (from `sturcture_list` or more) and logging
- [ ] Remove redundant keys in `dataload.py` (node_features, pos, cell, cell_shift, num_atoms(?), per_atom_energy)

**Comment by thangckt** [comment]
I suggest refactor the code to process dataset. It too complicate, make somebody lazy to contribute

**Comment by thangckt** [comment]
hi @YutackPark 
Are you consider to pure `pyproject.toml` ? I see many other project do not need `setup.*` files.
I also recommand to use `ruff` as both linter & formatter, it demonstrates better performance over other ones.

**Comment by YutackPark** [comment]
Hi @thangckt 
> Are you consider to pure pyproject.toml ? I see many other project do not need setup.* files.
Absolutely yes. But I left `setup.py` for editable mode when I first wrote `project.toml`. Because I wasn't able to install SevenNet with `pip -e`.

> I also recommand to use ruff as both linter & formatter, it demonstrates better performance over other ones.
Thanks for the recommendation. Maybe I could migrate to `ruff` with #98, and before github actions.

**Comment by YutackPark** [comment]
Close as remaining changes are not planned in short term
